torch_xla/experimental/distributed_checkpoint/manager.py

```diff
@@ -175,7 +201,12 @@ def save_async(self,
     Returns:
       True if a checkpoint was taken and False otherwise.
     """
-    raise NotImplementedError
+    if self.should_save(step) or force:
+      # Move the state_dict to CPU
+      cpu_state_dict = _sharded_cpu_state_dict(state_dict)
+      self._async_queue.put((step, cpu_state_dict))
```

Comment:
is this blocking when the size exceeds or at the capacity?

Reply:
Calls to `put` will block when the queue is at capacity